### DISCUSSION TITLE
Do you welcome new integrations?

### DISCUSSION THREAD

**Discussion by binwiederhier** [healthchecks/healthchecks#590] [Q&A]
I see a lot of [open issues](https://github.com/healthchecks/healthchecks/issues?q=is%3Aissue+is%3Aopen+label%3A%22new+integration%22) for new integrations. If I were to add my own service (https://ntfy.sh) as an integration (i.e. contribute code), would that be welcome, or do you think you have enough integrations?

**Comment by cuu508** [comment]
Hi @binwiederhier, ntfy.sh looks nifty! I like the simplicity, and the openness of it, kudos!

I'm open to adding new integrations, my main criteria for deciding what to add is **end-user interest** (across Github issues, private emails, tweets, etc.). Some other aspects are:

* would it be fun to work on? (let's be honest ;-)
* how much maintenance burden it would add – is the API good, are developer docs good, will I need a new trial account every time I test the integration
* Does the new integration enable something that is otherwise not possible (or is very inconvenient) via webhooks or email? 

On the last point, it looks like ntfy has awesomely-simple API that is easy to use via webhook integration:

![image](https://user-images.githubusercontent.com/661859/145542722-282a3e2d-7add-4024-bd22-abe0fed8b903.png)

I would not make a custom integration for ntfy myself (no end users have asked for it, and webhooks work fine). But if you make a complete PR, I would accept, because it seems  the extra maintenance burden would be low to none.

**Comment by binwiederhier** [comment]
Thank you for getting back to me @cuu508, I appreciate the thorough response. And you even tried it :-)

> end-user interest 

I acknowledge that other than myself, there is no interest right now. I am obviously trying to build it out, including integrations into tools such as healthchecks.io. So I wouldn't be offended if you said no :-)

> Does the new integration enable something that is otherwise not possible (or is very inconvenient) via webhooks or email?

Good call on the webhooks integration. Nice that it just works. 

> But if you make a complete PR, I would accept, because it seems the extra maintenance burden would be low to none. 

That is fantastic to hear. I'll create a PR in the next couple of days, including the docs and all that. I created this ticket for it: https://github.com/binwiederhier/ntfy/issues/49

**Reply by cuu508** [reply]
I've now added a "native" integration – https://github.com/healthchecks/healthchecks/issues/728

Examples of notifications in the app:

![image](https://user-images.githubusercontent.com/661859/203799335-22fac005-398b-4819-aee0-a187b923a3ef.png)

**Reply by binwiederhier** [reply]
Awesome!!
